cgen error for using a `default := p or { T{} }` construct in a generic function

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.7 f61a4f1.d5c2ebc
OS: linux, Ubuntu 20.04.6 LTS
Processor: 2 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /space/v/oo
vexe: /space/v/oo/v
vexe mtime: 2024-09-04 19:52:03

vroot: OK, value: /space/v/oo
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.46.0
Git vroot status: weekly.2024.36-8-gd5c2ebce
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg a.v`
```v
pub fn f[T](defaults ?T) {
	default := defaults or { T{} }
	dump(default)
}	
		 
f(123)

```

**What did you expect to see?**

a compiled program

**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/a.01J6Z9GC8NRPDKZDBDK9QGBFNR.tmp.c:13299: error: switch expected
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -25,7 +25,7 @@ fn (mut g Gen) expr_with_opt_or_block(expr ast.Expr, expr_typ ast.Type, var_expr
 		} else {
 			'${expr}'
 		}
-		g.writeln('if (${expr_var}.state != 0) { // assign')
+		g.writeln('if (${c_name(expr_var)}.state != 0) { // assign')
 		if expr is ast.Ident && expr.or_expr.kind == .propagate_option {
 			g.writeln('\tpanic_option_not_set(_SLIT("none"));')
 		} else {
@@ -37,7 +37,7 @@ fn (mut g Gen) expr_with_opt_or_block(expr ast.Expr, expr_typ ast.Type, var_expr
 			// handles stmt block which returns something
 			// e.g. { return none }
 			if stmts.len > 0 && stmts.last() is ast.ExprStmt && stmts.last().typ != ast.void_type {
-				g.gen_or_block_stmts(var_expr.str(), '', stmts, ret_typ, false)
+				g.gen_or_block_stmts(c_name(var_expr.str()), '', stmts, ret_typ, false)
 			} else {
 				// handles stmt block which doesn't returns value
 				// e.g. { return }

--- a/vlib/v/tests/generics/generic_options_with_reserved_ident_test.v
+++ b/vlib/v/tests/generics/generic_options_with_reserved_ident_test.v
@@ -1,0 +1,15 @@
+pub fn f[T](defaults ?T) T {
+	default := defaults or { T{} }
+	dump(default)
+	return default
+}
+
+fn test_generic_options_with_reserved_ident() {
+	ret1 := f(123)
+	println(ret1)
+	assert ret1 == 123
+
+	ret2 := f('hello')
+	println(ret2)
+	assert ret2 == 'hello'
+}

